custom/src/main/java/io/lumigo/javaagent/common/SecretScrubber.java

```
@@ -0,0 +1,103 @@
+  Logger LOGGER = Logger.getLogger(SecretScrubber.class.getName());
+
+  JsonFactory JSON_FACTORY = new JsonFactory();
+
+  public static final int DEFAULT_ATTRIBUTE_VALUE_LENGTH_LIMIT = 2048;
+  public static final String LUMIGO_SECRET_MASKING_REGEX = "lumigo.secret.masking.regex";
+  public static final String LUMIGO_SECRET_MASKING_ALL_MAGIC = "all";
+  public static final String SCRUBBED_VALUE = "****";
+
+  public static final List<String> DEFAULT_REGEX_KEYS =
+      Arrays.asList(
+          ".*pass.*",
+          ".*key.*",
+          ".*secret.*",
+          ".*credential.*",
+          ".*passphrase.*",
+          ".*token.*",
+          "SessionToken",
+          "x-amz-security-token",
+          "Signature",
+          "Credential",
+          "Authorization");
+
+  ParseExpressionResult parseExpressions(ConfigProperties config, String overrideEnvName) {
+    final String sourceEnvVar;
+    if (!Strings.isBlank(overrideEnvName) && !Strings.isBlank(config.getString(overrideEnvName))) {
+      sourceEnvVar = overrideEnvName;
+    } else if (config.getString(LUMIGO_SECRET_MASKING_REGEX) != null) {
+      sourceEnvVar = LUMIGO_SECRET_MASKING_REGEX;
+    } else {
+      sourceEnvVar = "";
+    }
+
+    final String regExps = !sourceEnvVar.isEmpty() ? config.getString(sourceEnvVar) : "";
+    assert regExps != null;
+    if (regExps.equalsIgnoreCase(LUMIGO_SECRET_MASKING_ALL_MAGIC)) {
+      return new ParseExpressionResult(sourceEnvVar, regExps);
+    }
+
+    final List<Pattern> patterns;
+    if (!Strings.isBlank(regExps)) {
+      patterns = new ArrayList<>();
+      try (JsonParser parser = JSON_FACTORY.createParser(regExps)) {
+        if (!parser.nextToken().isStructStart()
+            || !JsonToken.START_ARRAY.equals(parser.currentToken())) {
+          throw new IllegalArgumentException();
+        }
+        while (!JsonToken.END_ARRAY.equals(parser.nextToken())) {
+          if (parser.currentToken().equals(JsonToken.VALUE_STRING)) {
```

If it's not a string, we should fall back to the default secret scrubbing patterns. If I read the code right, now we would silently skip the horror.